controllers/referee/referee.py

```diff
@@ -1,120 +1,133 @@
-            print('performance:1' if CI else 'Red wins coverage: %s >= %s' % (self.coverage[0], self.coverage[1]))
-            print('Red wins coverage: %s >= %s' % (self.coverage[0], self.coverage[1]))
-        else:
-            print('performance:0' if CI else 'Blue wins coverage: %s > %s' % (self.coverage[1], self.coverage[0]))
-            print('Blue wins coverage: %s > %s' % (self.coverage[1], self.coverage[0]))
-
-
-# create the referee instance and run main loop
-CI = os.environ.get("CI")
-referee = Referee()
-referee.init()
-referee.run(CI)
-if CI:
-    referee.simulationSetMode(referee.SIMULATION_MODE_PAUSE)
+# Copyright 1996-2023 Cyberbotics Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Referee supervisor controller for the Robot Wrestling Tournament."""
+
+import math
+import os
+from controller import Supervisor, AnsiCodes
+
+
+class Referee (Supervisor):
+    def init(self):
+        # create an array of size [3][10] filled in with zeros
+        self.digit = [[0] * 10 for i in range(3)]
+        for j in range(3):
+            for i in range(10):
+                self.digit[j][i] = self.getDevice('digit ' + str(j) + str(i))
+        self.current_digit = [0, 0, 0]  # 0:00
+        self.robot = [0] * 2
+        self.robot[0] = self.getFromDef('WRESTLER_RED')
+        self.robot[1] = self.getFromDef('WRESTLER_BLUE')
+        self.min = [[0] * 3 for i in range(2)]
+        self.max = [[0] * 3 for i in range(2)]
+        for i in range(2):
+            self.min[i] = self.robot[i].getPosition()
+            self.max[i] = self.robot[i].getPosition()
+        self.coverage = [0] * 2
+        self.ko_count = [0] * 2
+        # linear motors on the side of the ring to display the coverage visually
+        self.indicator = [0] * 2
+        self.indicator[0] = self.getDevice('red indicator')
+        self.indicator[1] = self.getDevice('blue indicator')
+
+    def display_time(self, minutes, seconds):
+        for j in range(3):
+            self.digit[j][self.current_digit[j]].setPosition(1000)  # far away, not visible
+        self.current_digit[0] = minutes
+        self.current_digit[1] = seconds // 10
+        self.current_digit[2] = seconds % 10
+        for j in range(3):
+            self.digit[j][self.current_digit[j]].setPosition(0)  # visible
+
+    def run(self, CI):
+        # Performance output used by automated CI script
+        game_duration = 3 * 60 * 1000  # a game lasts 3 minutes
```

Comment:
In the [README file](https://github.com/cyberbotics/wrestling/blob/main/README.md?plain=1#L92) we say that a match is 5 minutes long. I don't know if we want to increase the time limit in the code or if we want to change the README to be the same as the code; 3 minutes seems enough for me.

Reply:
3 minutes looks for me as well. I will modify the README accordingly.